types.go

```diff
@@ -42,47 +52,52 @@ var (
 //      T *Thingy `optional:"true"`
 //    }
 //
-type In struct{}
+type In struct{ digSentinel }
```

Comment:
To clarify: digSentinel isn't used at all but I want dig.In and dig.Out as having unexported or private fields rather than a plain `struct{}`.